src/pathwayvis/components/map/map.component.ts

```diff
@@ -413,16 +413,28 @@ class MapComponentCtrl {
     if (selected.method.id === 'fva' || selected.method.id === 'pfba-fva') {
 
       // const fvaData = reactionData;
-      const fvaData = _.pickBy(reactionData, (d) => Math.abs((d.upper_bound + d.lower_bound) / 2) > 1e-7);
-
-      reactionData = _.mapValues(fvaData, (d) => (d.upper_bound + d.lower_bound) / 2);
-
+      const fvaData = reactionData;
+      for (const i in fvaData) {
+        if (Math.abs((fvaData[i].upper_bound + fvaData[i].lower_bound) / 2) < 1e-7) {
+          delete fvaData[i];
+        }
+      }
+      // const fvaData = _.pickBy(reactionData, (d) => Math.abs((d.upper_bound + d.lower_bound) / 2) > 1e-7);
+      reactionData = {}; // clean the object before insert the new values.
+      Object.keys(fvaData).map((key, index) =>
+        reactionData[key] = (fvaData[key].upper_bound + fvaData[key].lower_bound) / 2);
```

Comment:
Please don't modify variables inside a `map`. If you have to do this, better use `forEach`. but this is not a good case for that. Please check `Object.entries`, `Object.assign`. If you need some guidance let me know.